api/src/services/authentication.ts

```diff
@@ -1,27 +1,25 @@
-import argon2 from 'argon2';
 import jwt from 'jsonwebtoken';
 import { Knex } from 'knex';
 import ms from 'ms';
 import { nanoid } from 'nanoid';
-import { authenticator } from 'otplib';
 import getDatabase from '../database';
 import emitter, { emitAsyncSafe } from '../emitter';
 import env from '../env';
-import {
-	InvalidCredentialsException,
-	InvalidOTPException,
-	InvalidPayloadException,
-	UserSuspendedException,
-} from '../exceptions';
+import auth from '../auth';
+import { DEFAULT_AUTH_PROVIDER } from '../constants';
+import { InvalidCredentialsException, User } from '@directus/auth';
+import { InvalidOTPException, UserSuspendedException } from '../exceptions';
 import { createRateLimiter } from '../rate-limiter';
-import { ActivityService } from '../services/activity';
+import { ActivityService } from './activity';
+import { TFAService } from './tfa';
 import { AbstractServiceOptions, Accountability, Action, SchemaOverview, Session } from '../types';
 import { SettingsService } from './settings';
 import { merge } from 'lodash';
 
 type AuthenticateOptions = {
-	email: string;
+	identifier: string;
```

Comment:
I just realized this is a breaking change. I think we should map `email = identifier` before firing the `auth` hook and mark the email field as depreciated in the docs, warning users that email may not always represent an email value.

Thoughts @rijkvanzanten?